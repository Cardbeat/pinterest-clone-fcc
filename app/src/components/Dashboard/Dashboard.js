import React from "react";

const Dashboard = () => {
  return <>this is the dashboard</>;
};

export default Dashboard;
