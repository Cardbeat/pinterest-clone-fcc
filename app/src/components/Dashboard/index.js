import React from "react";

// using masonry,js to display image like pinterest
// https://github.com/tsuyoshiwada/react-stack-grid

const Dashboard = () => {
  return <>this is the dashboard</>;
};

export default Dashboard;
