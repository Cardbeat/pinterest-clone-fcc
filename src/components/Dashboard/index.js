import React, {useEffect, useState} from 'react';
import Grid from '../Grid';
import {fetchData} from '../../firebase';
import ReactLoading from 'react-loading';
import styled from 'styled-components';

const LoadingContainer = styled.div`
  width: 50%;
  margin: 0 auto
  padding: 20px
`;

export default function Dashboard() {
  const [Doc, setDoc] = useState(false);

  useEffect(() => {
    fetchData().then(doc => setDoc(doc));
  }, []);

  return (
    <>
      {Doc ? (
        Doc.map((user, index) => {
          console.log(user.data.name);
          return (
            <Grid
              key={index}
              name={user.data.name}
              token={user.id}
              images={user.data.images}
            />
          );
        })
      ) : (
        <LoadingContainer>
          <ReactLoading type={'spin'} height={'10%'} width={'10%'} />
        </LoadingContainer>
      )}
    </>
  );
}
