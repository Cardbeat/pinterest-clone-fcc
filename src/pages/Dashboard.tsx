import React from "react";

const Dashboard: React.FC = () => {
  return <p> this is the dashboard</p>;
};

export default Dashboard;
