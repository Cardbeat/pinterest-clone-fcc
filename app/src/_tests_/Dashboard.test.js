import React from "react";
import { render, cleanup } from "@testing-library/react";
import Dashboard from "../components/Dashboard/index";
afterEach(cleanup);

test("render component ", () => {
  const { asFragment } = render(<Dashboard />);
  expect(asFragment()).toMatchSnapshot();
});
