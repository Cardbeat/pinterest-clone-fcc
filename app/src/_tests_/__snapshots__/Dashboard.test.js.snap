// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render component  1`] = `
<DocumentFragment>
  this is the dashboard
</DocumentFragment>
`;
